mne/decoding/gsoc.py

```diff
@@ -0,0 +1,522 @@
+            The kind of event to apply. if None, a dict of inst will be return
+            one for each type of event xdawn has been fitted.
+        include : array_like of int | None (default None)
+            The indices referring to columns in the ummixing matrix. The
+            components to be kept. If None, the first n_components (as defined
+            in the Xdawn constructor) will be kept.
+        exclude : array_like of int | None (default None)
+            The indices referring to columns in the ummixing matrix. The
+            components to be zeroed out. If None, all the components except the
+            first n_components will be exclude.
+
+        Returns
+        -------
+        out : dict of instance
+            A dict of instance (from the same type as inst input) for each
+            event type in event_id.
+        """
+
+        if not isinstance(X, np.ndarray):
+            raise ValueError("Given data should be numpy array, got "
+                             "%s instead" % type(X))
+
+        data = np.hstack(X.reshape(X.shape[0], self.n_chan, X.shape[1] /
+                                   self.n_chan))
+        data_dict = dict()
+        if event_id is None:
+            event_id = self.event_id
+
+        for eid in event_id:
+
+            data_r = self._pick_sources(data, include, exclude, eid)
+            data_r = np.array(np.split(data_r, self.n_chan, 1))
+            data_dict[eid] = data_r
+
+        return data_dict
+
+    def apply(self, inst, event_id=None, include=None, exclude=None):
+        """apply in this version of xdawn is not supported.
+
+        Parameters
+        ----------
+        inst : instance of Raw | Epochs | Evoked
+            The data to be processed.
+        event_id : dict | list of str | None (default None)
+            The kind of event to apply. if None, a dict of inst will be return
+            one for each type of event xdawn has been fitted.
+        include : array_like of int | None (default None)
+            The indices referring to columns in the ummixing matrix. The
+            components to be kept. If None, the first n_components (as defined
+            in the Xdawn constructor) will be kept.
+        exclude : array_like of int | None (default None)
+            The indices referring to columns in the ummixing matrix. The
+            components to be zeroed out. If None, all the components except the
+            first n_components will be exclude.
+
+        Returns
+        -------
+        None
+        """
+
+        warnings.warn("Warning, apply is not supported in this version of "
+                      "xdawn. Use inverse_transform method instead.")
+
+        pass
+
+
+class Vectorizer(TransformerMixin):
+    """Class to chain mne transformer output to scikit-learn estimators.
+
+    MNE transformer have three, four dimensional output. This class converts
+    them into two dimension so as to comply with scikit-learn API.
+    """
+
```

Comment:
Ok we may think about the name later then. NDVectorizer or ArrayVectorizer or which ever name is more helpful.